kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -511,6 +516,36 @@ def kube_job_status_succeeded(self, metric, scraper_config):
                     tags.append(self._format_tag(label_name, label_value, scraper_config))
             self.job_succeeded_count[frozenset(tags)] += sample[self.SAMPLE_VALUE]
 
+    def kube_job_status_start_time(self, metric, scraper_config):
+        for sample in metric.samples:
+            tags = [] + scraper_config['custom_tags']
+            for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
+                tags.append(self._format_tag(label_name, label_value, scraper_config))
+            self.job_start_time[frozenset(tags)] += int(sample[self.SAMPLE_VALUE])
+
+    def kube_job_status_completion_time(self, metric, scraper_config):
+        for sample in metric.samples:
+            tags = [] + scraper_config['custom_tags']
+            for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
+                tags.append(self._format_tag(label_name, label_value, scraper_config))
+            self.job_completion_time[frozenset(tags)] += int(sample[self.SAMPLE_VALUE])
+
+    def kube_job_status_execution_time(self, scraper_config):
+        metric_name = scraper_config['namespace'] + '.job.execution_time'
+        for job_tags, completion_time in iteritems(self.job_completion_time):
+            if job_tags in self.job_start_time:
```

Comment:
AFAICT it's always true because you have a `defaultdict`.

Reply:
it is true?
I tried writing the following code as a trial, but it did not have such a result.
```
from collections import Counter, defaultdict
from six import iteritems

start_time = defaultdict(int)
end_time = defaultdict(int)
tags = ['a','b','c']
frozen_tag = frozenset(tags)
start_time[frozenset(tags)] = 10

print(frozen_tag in start_time) #True
print(frozen_tag in end_time) #False

end_time[frozenset(tags)] = 20
print(frozen_tag in end_time) #True
```

By the way, I put it in order to solve this part
>If calculated on the graph, only kube_job_status_start_time will be sent if the job is running, so it will be negative values when calculated.

Reply:
You're right! So I think there may be something weird here. Do we want to increase the start time? Presumably we get a different job every time, because adding start_time doesn't make much sense, does it? I think making it regular dicts would be better.

Reply:
Thank you! As there was no reason to use defaultdict in particular